testflinger_agent/job.py

```diff
@@ -254,24 +264,19 @@ def cleanup(signum, frame):
                     f.write(buf)
                     process.kill()
                     break
-                # Don't spam the server, only flush the buffer if there
-                # is output and it's been more than 10s
-                if live_output_buffer and time.time() - buffer_timeout > 10:
-                    buffer_timeout = time.time()
-                    # Try to stream output, if we can't connect, then
-                    # keep buffer for the next pass through this
-                    if self.client.post_live_output(
-                        self.job_id, live_output_buffer
-                    ):
-                        live_output_buffer = ""
-            buf = process.stdout.read()
-            if buf:
-                buf = buf.decode(sys.stdout.encoding, errors="replace")
-                sys.stdout.write(buf)
-                live_output_buffer += buf
-                f.write(buf)
+
+                # Check if job was canceled
+                if (
+                    self.client.check_job_state(self.job_id) == "cancelled"
+                    and self.phase != "provision"
```

Comment:
Is it worth setting these as globals somewhere you can import/pull from to use them every time?

Reply:
That's a fair question, but to me, the code smell is worse than that. I'm thinking that all these special cases tell me that I should look towards reorganizing this in a bigger way so that each phase has it's own object, or possibly even a subclass with all the special handling that it needs to do. For now, phases are necessarily just strings, but some day it may be a slightly more formal definition.

Reply:
and thanks a LOT for the review!